File: app/src/main/java/com/solt9029/editmasterandroid/repository/ScoreRepository.java
package com.solt9029.editmasterandroid.repository;

import com.solt9029.editmasterandroid.model.Score;
import com.solt9029.editmasterandroid.service.ScoreService;
import io.reactivex.Single;

import javax.inject.Inject;
import java.util.List;

public class ScoreRepository {
    private ScoreService service;

    @Inject
    public ScoreRepository(ScoreService service) {
        this.service = service;
    }

    public Single<List<Score>> getScoreTimeline(Integer count, String keyword, Integer maxId, Integer sinceId) {
        return service.getScoreTimeline(count, keyword, maxId, sinceId);
    }

    public Single<Score> getScore(Integer id) {
        return service.getScore(id);
    }
}

File: app/src/main/java/com/solt9029/editmasterandroid/repository/ScoreRepository.kt
package com.solt9029.editmasterandroid.repository

import com.solt9029.editmasterandroid.entity.Score
import com.solt9029.editmasterandroid.service.ScoreService
import io.reactivex.Single
import retrofit2.Response

import javax.inject.Inject

class ScoreRepository @Inject constructor(private val service: ScoreService) {
    fun getScoreTimeline(count: Int?, keyword: String?, maxId: Int?, sinceId: Int?): Single<List<Score>> {
        return service.getScoreTimeline(count, keyword, maxId, sinceId)
    }

    fun getScore(id: Int?): Single<Score> {
        return service.getScore(id)
    }

    fun createScore(score: Score): Single<Response<Score>> {
        return service.createScore(score)
    }
}
